### DISCUSSION TITLE
BDBrowser 1.9.5.20231111 - Launch Error

### DISCUSSION THREAD

**Discussion by DeadLyBro** [tsukasa/BdBrowser#31] [Problems and issues]
When I open Discord website, I see this error and I can't see any settings on setting page.  

Error: Uncaught (in promise) TypeError: modules_discordmodules__WEBPACK_IMPORTED_MODULE_5__.Z.GuildStore?.getGuilds is not a function. 
Lines: frontend.js:2112:219 and frontend.js:6685:10

This discussion may be related other discussions but I think they doesn't understand what is causing this. 
So don't delete this post and if you can merge with other discussions its fine.

I can see this but there is not option for BD; 
BDBrowser 1.9.5.20231111
betterdiscord.asar (v1.9.6, VFS)

**Comment by DeadLyBro** [comment]
Plus if I try to open Plugin repo in Previous Version (BDBrowser 1.9.5.20231101) I recieve error like this: 
![Screenshot_1](https://github.com/tsukasa/BdBrowser/assets/31214077/ab01f88b-35e7-4f36-8e1e-bdd470b006bc)
It won't open, just lagging the tab.

**Comment by tsukasa** [comment]
Greetings,

thank you for taking the time to report this! You did indeed spot a bug that made it into the 20231111 release and was consequently fixed in commit c4d859c31f29e2f13dc99b2af7b8ca6bdf4c53a1. Since the bug does not cause any regressions and I am still trying to build a less janky solution (as described [here](https://github.com/tsukasa/BdBrowser/discussions/29#discussioncomment-7540025)), I did not issue a release yet.

You can grab the [build artifact from the last commit](https://github.com/tsukasa/BdBrowser/actions/runs/6834364900) to mitigate this, in case you want to make sure this has no adverse effects and/or fixes your issue.

BetterDiscord should load properly either way. If that is not the case, please create a VFS backup, format the VFS and give a fresh, empty version of BetterDiscord with no plugins a try. In case BetterDiscord does load after a VFS format, I would advise to restore the backup, disable the plugins via the extension options and check if any of them offer updates after reloading the tab. Malfunctioning plugins might be the cause.

If you are not using the stable web client but canary, you also might run into unexpected results.

As for the BDFDB log entries: I cannot replicate this on my end (regular "stable" webclient) and therefore can only repeat the "try updating the plugins and be on stable". You can also give a hard reload a try to ensure the preload was processed properly.

**Reply by DeadLyBro** [reply]
For the BDFDB log entries error, I'm sure I'm using latest version all the things.

**Reply by DeadLyBro** [reply]
When I login first time, BD loaded succesfully on user settings page but if I refresh the page, I can't see again... 
Screenshot: 
![Screenshot_1](https://github.com/tsukasa/BdBrowser/assets/31214077/ce9c7560-e678-4295-afcb-6d2e6d0d3eb1)
Edit: I updated extension to [that artifact](https://github.com/tsukasa/BdBrowser/actions/runs/6834364900) and results like this.

**Reply by DeadLyBro** [reply]
This is the last logs for BD: 
![Screenshot_2](https://github.com/tsukasa/BdBrowser/assets/31214077/e8183f49-2ebd-4dae-a11d-2878d23dce2a)
I can't see any error on devtools or extensions page.

**Reply by tsukasa** [reply]
Thank you for the additional details. From the given information, it is reasonable to assume that this is what currently happens:

- The changes in BDBrowser 20231111 and newer affect when BetterDiscord is injected.
- From the details visible in the logs, the callback gets registered but does not fire - thus not injecting BetterDiscord's renderer.
- I assume that the initialization (1603ms) takes too long, and the "window of opportunity" within the web application passes. The code that would work around this was removed in 20231111.
- This would explain why 20231101 seems to produce better results for you; the now-removed code path comes in to perform the initialization.

Right now, I would amend my previous recommendation. Instead of trying the artifact from the build, it would be better to roll back to 20231101 in your case. 20231101 should provide a more stable/reliable start-up, as you have described in your initial message.

As for the plugin, my previous recommendation stands. I did give the Plugin Repository plugin a quick before writing my last message and could not reproduce the warnings, thus making it rather hard to respond to.

**Reply by DeadLyBro** [reply]
Thank you for quick reply. I'm returning to previous build.
